builtin/providers/aws/resource_aws_glacier_vault.go

```diff
@@ -163,8 +163,14 @@ func resourceAwsGlacierVaultRead(d *schema.ResourceData, meta interface{}) error
 	if awserr, ok := err.(awserr.Error); ok && awserr.Code() == "ResourceNotFoundException" {
 		d.Set("access_policy", "")
 	} else if pol != nil {
-		policy, _ := normalizeJsonString(*pol.Policy.Policy)
-		d.Set("access_policy", policy)
+		policy, err := normalizeJsonString(*pol.Policy.Policy)
+		if err != nil {
+			return fmt.Errorf("access policy contains an invalid JSON: %s", policy)
+		}
+
+		if err := d.Set("access_policy", policy); err != nil {
```

Comment:
I am not sure what is the convention, to be honest. I am simply following the pattern I saw amongst other files, as per:

```
$ ack 'if err := d.Set\(' -l | wc -l
      61
```

I will remove this, since it is indeed just a string.